### PR TITLE
Correct CDX API timestamps with month of 00

### LINKED ISSUE #88: ValueError: time data
I happened to be doing this:

```python
from wayback import WaybackClient

ia = WaybackClient()
for result in ia.search('lapdonline.org', matchType='prefix'):
    print(result)
```

and noticed that after running for 10 minutes or so it blew up with:

```python
Traceback (most recent call last):
  File "/Users/edsummers/Projects/wayback/wayback/_client.py", line 543, in search
    capture_time = _utils.parse_timestamp(data.timestamp)
  File "/Users/edsummers/Projects/wayback/wayback/_utils.py", line 57, in parse_timestamp
    .strptime(''.join(timestamp_chars), URL_DATE_FORMAT)
  File "/usr/local/Cellar/python@3.10/3.10.6_1/Frameworks/Python.framework/Versi
ons/3.10/lib/python3.10/_strptime.py", line 568, in _strptime_datetime
    tt, fraction, gmtoff_fraction = _strptime(data_string, format)
  File "/usr/local/Cellar/python@3.10/3.10.6_1/Frameworks/Python.framework/Versions/3.10/lib/python3.10/_strptime.py", line 349, in _strptime
    raise ValueError("time data %r does not match format %r" %
ValueError: time data '20000008241731' does not match format '%Y%m%d%H%M%S'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/edsummers/Projects/wayback/./x.py", line 6, in <module>
    for result in ia.search('lapdonline.org', matchType='prefix'):
  File "/Users/edsummers/Projects/wayback/wayback/_client.py", line 547, in search
    raise UnexpectedResponseFormat(
wayback.exceptions.UnexpectedResponseFormat: Could not parse CDX output: "org,lapdonline)/community/op_valley_bureau/north_hollywood/map/map.htm 20000008241731 http://www.lapdonline.org:80/community/op_valley_bureau/north_hollywood/map/map.htm text/html 200 2GPKQMU3BLZXOEZ5EWDQEYHPMKWEHNT3 1158" (query: {'url': 'lapdonline.org', 'matchType': 'prefix', 'showResumeKey': 'true', 'resolveRevisits': 'true'})
```

It looks like the CDX API returned a datetime `20000008241731` which throws an exception during parse because `00` isn't a valid month?

I don't know what the solution is here:

- ignore the record?
- see if the new CDX API is better behaved and switch to it?
- something else?

### PR DESCRIPTION
**Pull request by edsu**
This commit extends the existing logic for handling invalid days of `00` to months that are `00`. It also adds a warning to be logged in both situations.

So if a timestamp of `20200001120000` will get rewritten to `20200112000000` prior to conversion to a datetime.

I have tested on live CDX API data that was failing, and this fix causes the full result to be returned. If more information is known about why this approach is taken it would be good to add in a comment?

Closes #88

### CODE REVIEW

File: wayback/_utils.py
Comment (Mr0grog) [inline wayback/_utils.py:4]: Can you please also create a logger so people know *where* this warning is coming from if they hit it? e.g:

```py
logger = logging.getLogger(__name__)
```

**Review by Mr0grog — CHANGES_REQUESTED** [review]
This looks great, excepting one small tweak. Thanks for adding the warnings; I’m not sure why we didn’t think to do that originally.

> If more information is known about why this approach is taken it would be good to add in a comment?

Yes! Looking back at the original PR this adds to (#85), the existing comments could have been much clearer. Feel free to add to them, or I will before merging.

I’m going to wait on this a little so we can get some feedback from Internet Archive folks, since last time (#85) the right answer turned out not to be anything like what we expected.

**Comment by Mr0grog** [comment]
I'll also have to figure out what’s going on with the docs here. :\

File: wayback/_utils.py
Comment (edsu) [reply wayback/_utils.py:4]: Nice catch, logger is configured now.

**Comment by edsu** [comment]
I tried to address the docs build over in #90. I'm not sure if that's the right approach though, so of course please feel free to close if not.

**Comment by Mr0grog** [comment]
Heard back from Wayback folks here: https://internetarchive.slack.com/archives/C4PMRAN00/p1664039239024329

And it seems like our solution is the correct one for now. I should have time tomorrow (Sept. 29) to get this merged in (after we resolve the docs issue) and cut a release.

**Review by Mr0grog — APPROVED** [review]
I fixed the docs issue in #91, rebased this, and added a more explanatory comment in the code (as you suggested @edsu, thanks for prompting!).

This ought to be good to go, but I’ll wait to merge until tomorrow, when I have less tired eyes to double-check this with.

**Comment by Mr0grog** [comment]
@edsu FYI, this is released in v0.3.3!

**Comment by edsu** [comment]
Thank you @Mr0grog !

**Pull request merged**
